ui/components/ui/jazzicon/jazzicon.component.js

```diff
@@ -9,56 +9,39 @@ const iconFactory = iconFactoryGenerator(jazzicon);
  * Wrapper around the jazzicon library to return a React component, as the library returns an
  * HTMLDivElement which needs to be appended.
  */
-export default class Jazzicon extends PureComponent {
-  static propTypes = {
-    address: PropTypes.string.isRequired,
-    className: PropTypes.string,
-    diameter: PropTypes.number,
-    style: PropTypes.object,
-    tokenList: PropTypes.object,
-  };
 
-  static defaultProps = {
-    diameter: 46,
-  };
-
-  container = createRef();
-
-  componentDidMount() {
-    this.appendJazzicon();
-  }
-
-  componentDidUpdate(prevProps) {
-    const { address: prevAddress, diameter: prevDiameter } = prevProps;
-    const { address, diameter } = this.props;
-
-    if (address !== prevAddress || diameter !== prevDiameter) {
-      this.removeExistingChildren();
-      this.appendJazzicon();
-    }
-  }
-
-  removeExistingChildren() {
-    const { children } = this.container.current;
-
-    for (let i = 0; i < children.length; i++) {
-      this.container.current.removeChild(children[i]);
-    }
-  }
-
-  appendJazzicon() {
-    const { address, diameter, tokenList } = this.props;
-    const image = iconFactory.iconForAddress(
+function Jazzicon({
+  address,
+  className,
+  diameter = 46,
+  style,
+  tokenList = {},
+}) {
+  const image = useMemo(() => {
+    const imageNode = iconFactory.iconForAddress(
       address,
       diameter,
       tokenList[address.toLowerCase()],
     );
-    this.container.current.appendChild(image);
-  }
 
-  render() {
-    const { className, style } = this.props;
+    return imageNode.outerHTML;
+  }, [address, diameter, tokenList]);
 
-    return <div className={className} ref={this.container} style={style} />;
-  }
+  return (
+    <div
+      className={className}
+      style={style}
+      dangerouslySetInnerHTML={{ __html: image }}
```

Comment:
@brad-decker what's your thoughts on using `dangerouslySetInnerHTML` here?

Reply:
This isn't any more or less  safe than 'appendChild' with an 'image' that could literally be any html. If iconFactory.iconForAddress were to return malicious code we'd inject that malicious code. We *do* own that code so its less of a concern. 

I feel it would be better to append the element in the same way it was before just to avoid any additional unforeseen security implications of _dangerouslySetInnerHTML -- which doesn't yet exist in the codebase.